pkg/blueprint/partitioning_customizations_test.go

```diff
@@ -0,0 +1,671 @@
+				Btrfs: &blueprint.BtrfsCustomization{
+					Volumes: []blueprint.BtrfsVolumeCustomization{
+						{
+							Subvolumes: []blueprint.BtrfsSubvolumeCustomization{
+								{
+									Name:       "test",
+									Mountpoint: "/test",
+								},
+								{
+									Name:       "",
+									Mountpoint: "/test2",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedMsg: `btrfs subvolume with empty name in partitioning customizations`,
+		},
+		"boot-on-lvm": {
+			partitioning: &blueprint.PartitioningCustomization{
+				LVM: &blueprint.LVMCustomization{
+					VolumeGroups: []blueprint.VGCustomization{
+						{
+							LogicalVolumes: []blueprint.LVCustomization{
+								{
+									Name: "bewt",
+									FilesystemCustomization: blueprint.FilesystemCustomization{
+										Mountpoint: "/boot",
+									},
+								},
+								{
+									Name: "testlv",
+									FilesystemCustomization: blueprint.FilesystemCustomization{
+										Mountpoint: "/stuff2",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedMsg: `invalid mountpoint "/boot" for logical volume`,
+		},
+		"bootefi-on-lvm": {
+			partitioning: &blueprint.PartitioningCustomization{
+				LVM: &blueprint.LVMCustomization{
+					VolumeGroups: []blueprint.VGCustomization{
+						{
+							LogicalVolumes: []blueprint.LVCustomization{
+								{
+									Name: "bewtefi",
+									FilesystemCustomization: blueprint.FilesystemCustomization{
+										Mountpoint: "/boot/efi",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedMsg: `invalid mountpoint "/boot/efi" for logical volume`,
+		},
+		"boot-on-btrfs": {
+			partitioning: &blueprint.PartitioningCustomization{
+				Btrfs: &blueprint.BtrfsCustomization{
+					Volumes: []blueprint.BtrfsVolumeCustomization{
+						{
+							Subvolumes: []blueprint.BtrfsSubvolumeCustomization{
+								{
+									Name:       "test",
+									Mountpoint: "/test",
+								},
+								{
+									Name:       "bootbootboot",
+									Mountpoint: "/boot",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedMsg: `invalid mountpoint "/boot" for btrfs subvolume`,
+		},
+		"bootefi-on-btrfs": {
+			partitioning: &blueprint.PartitioningCustomization{
+				Btrfs: &blueprint.BtrfsCustomization{
+					Volumes: []blueprint.BtrfsVolumeCustomization{
+						{
+							Subvolumes: []blueprint.BtrfsSubvolumeCustomization{
+								{
+									Name:       "test",
+									Mountpoint: "/test",
+								},
+								{
+									Name:       "esp",
+									Mountpoint: "/boot/efi",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedMsg: `invalid mountpoint "/boot/efi" for btrfs subvolume`,
+		},
+	}
+
+	for name := range testCases {
+		tc := testCases[name]
+		t.Run(name, func(t *testing.T) {
+			err := tc.partitioning.ValidateSimple()
+			if tc.expectedMsg != "" {
+				assert.EqualError(t, err, tc.expectedMsg)
```

Comment:
Not sure the latter is better or worse.  If the expected message is long, it might be a distraction when the more important information is that an error wasn't returned, which you get in the `Error:` field.  When they don't match, you get both the expected and the actual in the `Error:` field, so repeating the expected message is extra noise (for long messages).